lib/cdo/contact_rollups.rb

```diff
@@ -427,19 +428,24 @@ def self.update_data_from_forms(form_kind)
 
       role = data['role_s']
 
-      # Skip if this form data has no info at all
-      next if street_address.nil? && city.nil? && state.nil? && postal_code.nil? && country.nil? && role.nil?
-
-      # Truncate all fields to 255 chars. We have some data longer than 255 chars but it is all garbage that somebody typed.
-      update_data = {}
-      update_data[:street_address] = street_address[0...255] unless street_address.nil?
-      update_data[:city] = city[0...255] unless city.nil?
-      update_data[:state] = state[0...255] unless state.nil?
-      update_data[:postal_code] = postal_code[0...255] unless postal_code.nil?
-      update_data[:country] = country[0...255] unless country.nil?
+      any_update = false
 
-      # add to batch to update
-      update_batch += conn[DEST_TABLE_NAME.to_sym].where(email: email).update_sql(update_data) + ";" unless update_data.empty?
+      # Skip if this form data has no info at all
+      if street_address.present? || city.present? || state.present? || postal_code.present? || country.present?
+        # If any geo input is present in form, update ALL geo fields including setting NULL values in DB for any that
+        # we don't have from this form. This will clear out any previous geo data from IP geo.
+        # Truncate all fields to 255 chars. We have some data longer than 255 chars but it is all garbage that somebody typed
+        update_data = {}
+        update_data[:street_address] = street_address.present? ? street_address[0...255] : nil
+        update_data[:city] = city.present? ? city[0...255] : nil
+        update_data[:state] = state.present? ? state[0...255] : nil
+        update_data[:postal_code] = postal_code.present? ? postal_code[0...255] : nil
+        update_data[:country] = country.present? ? country[0...255] : nil
```

Comment:
This can be simplified to:

```ruby
  update_data[:street_address] street_address[0...255] if street_address.present?
  update_data[:city] city[0...255] if city.present?
  #...
```

Then you can lose the pre-check on line 434 and do
```ruby
unless update_data.empty?
  # add to batch update ...
end
```

Then also you can remove the confusing `any_update` variable, and do
```ruby
next if update_data.empty? && role.blank?
```